### PR TITLE
Envoy 1.17 crashing with validation context secret discovery in ads mode on start up #15517

### PR DESCRIPTION
**Pull request by twghu**
Test to prevent crashing with validation context secret discovery in ads mode on start up

Signed-off-by: Tim Walsh <temporal.differential@gmail.com>

Commit Message: Test to prevent crashing with validation context secret discovery in ads mode on start up
Additional Description:
Risk Level:
Testing: test/server/server_test.cc
Docs Changes: 
Release Notes: 
It is not possible to configure any component of a primary cluster over ADS
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by htuch** [comment]
@adisuissa for first pass review.

**Review by adisuissa — CHANGES_REQUESTED** [review]
Thanks for working on this.
IIUC this PR is the same as #15809 (fixing #15517) merging into main.
The previous PR should probably be closed.

File: source/common/config/grpc_subscription_impl.cc [hunk 1]
```diff
@@ -36,6 +36,7 @@ void GrpcSubscriptionImpl::start(const absl::flat_hash_set<std::string>& resourc
     init_fetch_timeout_timer_->enableTimer(init_fetch_timeout_);
   }
 
+  RELEASE_ASSERT(grpc_mux_ != nullptr, "grpc_mux_ initialized with null value");
```
[the remark this answers is not in the document]
Comment (twghu) [reply source/common/config/grpc_subscription_impl.cc:39]: The next line immediately de-references the grpc_mux ptr, so will error if not checked prior.

`watch_ = grpc_mux_->addWatch(type_url_, resources, *this, resource_decoder_, options_);`

File: source/common/upstream/cluster_manager_impl.cc [hunk 2]
```diff
@@ -357,6 +356,11 @@ ClusterManagerImpl::ClusterManagerImpl(
           bootstrap.dynamic_resources().ads_config().set_node_on_first_message_only());
     }
   } else {
+    // Without a dynamic resource configuration (bootstrap.dynamic_resources() == null)
+    // an ads_mux == nullptr will cause an error in the
+    // SubscriptionFactoryImpl::subscriptionFromConfigSource() for the
+    // envoy::config::core::v3::ConfigSource::ConfigSourceSpecifierCase::kAds case.
+    // TODO can this condition be caught here?
```
[the remark this answers is not in the document]
Comment (twghu) [reply source/common/upstream/cluster_manager_impl.cc:363]: Ack, looking at call stack.

File: test/server/test_data/server/bad_sds_ads_config_source.yaml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+              ads: {}
+              resource_api_version: V3
+dynamic_resources:
+  ads_config:
+    api_type: GRPC
+    grpc_services:
+      envoy_grpc:
+        cluster_name: "ads_cluster"
+    transport_api_version: V3
+#node:
```
[the remark this answers is not in the document]
Comment (twghu) [reply test/server/test_data/server/bad_sds_ads_config_source.yaml:52]: Deleted

File: source/common/upstream/cluster_manager_impl.cc [hunk 4]
```diff
@@ -313,7 +313,6 @@ ClusterManagerImpl::ClusterManagerImpl(
       loadCluster(cluster, MessageUtil::hash(cluster), "", false, active_clusters_);
     }
   }
-
```
[the remark this answers is not in the document]
Comment (twghu) [reply source/common/upstream/cluster_manager_impl.cc:316]: tools/code_format/check_format.py fix removed it.

**Pull request closed without merging**